src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.service.MovieSessionService;
+
+@Service
+public class MovieSessionServiceImpl implements MovieSessionService {
+    @Inject
+    private MovieSessionDao movieSessionDao;
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        return movieSessionDao.add(movieSession);
+    }
+
+    @Override
+    public MovieSession get(Long id) {
+        return movieSessionDao.get(id).orElseThrow(
+                () -> new RuntimeException("Couldn't get a movie session"));
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        return movieSessionDao.getAll(movieId, date);
+    }
```

Comment:
The current implementation of `findAvailableSessions` in the service layer might not be performing optimally. According to the checklist, you should not get all `MovieSession` from dao and then sort them on the service layer. It's better for performance to extract 'ready to use' data from the DB on the dao layer. Ensure that the `MovieSessionDao` implementation of `getAll` method is optimized to filter sessions directly within the database query, especially to limit movie sessions within the provided date.